hyperspy/io.py

```diff
@@ -482,35 +487,39 @@ def load_with_reader(
         reader,
         signal_type=None,
         convert_units=False,
+        load_original_metadata=True,
         **kwds
     ):
     """Load a supported file with a given reader."""
     lazy = kwds.get('lazy', False)
     file_data_list = reader.file_reader(filename, **kwds)
-    objects = []
+    signal_list = []
 
     for signal_dict in file_data_list:
         if 'metadata' in signal_dict:
             if "Signal" not in signal_dict["metadata"]:
                 signal_dict["metadata"]["Signal"] = {}
             if signal_type is not None:
                 signal_dict['metadata']["Signal"]['signal_type'] = signal_type
-            objects.append(dict2signal(signal_dict, lazy=lazy))
+            signal = dict2signal(signal_dict, lazy=lazy)
             folder, filename = os.path.split(os.path.abspath(filename))
             filename, extension = os.path.splitext(filename)
-            objects[-1].tmp_parameters.folder = folder
-            objects[-1].tmp_parameters.filename = filename
-            objects[-1].tmp_parameters.extension = extension.replace('.', '')
+            signal.tmp_parameters.folder = folder
+            signal.tmp_parameters.filename = filename
+            signal.tmp_parameters.extension = extension.replace('.', '')
             if convert_units:
-                objects[-1].axes_manager.convert_units()
+                signal.axes_manager.convert_units()
+            if not load_original_metadata:
```

Comment:
It would have been better to modify ``dict2signal()`` in order to load ``original_metadata`` into a DTB. However, that's not trivial, so this is a good first step. Maybe we could open an issue about it not to forget it?

Reply:
Yes, I agree and there is already an open PR on this actually: https://github.com/hyperspy/hyperspy/pull/2092!

Reply:
Good point!